testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -13824,6 +13824,63 @@ void testTimestampDiff(boolean coercionEnabled) {
     f.checkNull("date_add(CAST(NULL AS DATE), interval 5 day)");
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6527">[CALCITE-6527]
+   * Add DATE_ADD function (enabled in Spark library)</a>.
+   */
+  @Test void testDateAddSpark() {
+    final SqlOperatorFixture f0 = fixture()
+        .setFor(SqlLibraryOperators.DATE_ADD_SPARK);
+    f0.checkFails("^date_add(date '2008-12-25', "
+            + "5)^",
+        "No match found for function signature "
+            + "DATE_ADD\\(<DATE>, <NUMERIC>\\)", false);
+
+    final SqlOperatorFixture f = f0.withLibrary(SqlLibrary.SPARK);
+    f.checkScalar("date_add(date '2016-02-22', 2)",
+        "2016-02-24",
+        "DATE NOT NULL");
+    f.checkScalar("date_add(date '2016-02-28', 2)",
+        "2016-03-01",
+        "DATE NOT NULL");
+    f.checkScalar("date_add(date '2016-03-01', -2)",
+        "2016-02-28",
+        "DATE NOT NULL");
+    f.checkScalar("date_add(date '2016-03-01', -2)",
+        "2016-02-28",
+        "DATE NOT NULL");
+    f.checkScalar("date_add(date '2016-03-01', -2.0)",
+        "2016-02-28",
+        "DATE NOT NULL");
+    f.checkScalar("date_add(date '2016-02-28', 2.0)",
+        "2016-03-01",
+        "DATE NOT NULL");
+    f.checkScalar("date_add(date '2016-02-28', '2.0')",
+        "2016-03-01",
+        "DATE NOT NULL");
+    f.checkScalar("date_add(timestamp '2016-02-22 13:00:01', '-2.0')",
+        "2016-02-20",
+        "DATE NOT NULL");
+    f.checkScalar("date_add(timestamp '2016-02-22 13:00:01', -2)",
+        "2016-02-20",
+        "DATE NOT NULL");
+    f.checkScalar("date_add(timestamp '2016-02-22 13:00:01', -2.0)",
+        "2016-02-20",
+        "DATE NOT NULL");
+    f.checkScalar("date_add(date '0001-01-01', '-2.0')",
+        "0000-12-30",
+        "DATE NOT NULL");
+    f.checkScalar("date_add(date '0001-01-01', '-367')",
```

Comment:
what is going on here? Why is `000/` a valid year?

Reply:
@mihaibudiu I raised a jira to discuss this issue. Should I delete this test?
https://issues.apache.org/jira/browse/CALCITE-6536

Reply:
I would still like to know why this test passes.

Reply:
If a test passes that shouldn't it means that there's somewhere another bug that is masking a problem.

Reply:
> If a test passes that shouldn't it means that there's somewhere another bug that is masking a problem.

agree. 
I will fix it, the date function does validation when it handles 000/or 0000 as a parameter, but when 000/or 0000 is returned as a result, no such validation is done, I think this is the bug

Reply:
@mihaibudiu I'm not sure how to solve this bug, it looks a bit complicated to me 😭

Reply:
I don't think Calcite can offer exact compatibility to Spark in this case.
The best solution for Calcite is to give a runtime error.
But I would still like to know why the date with a slash didn't report a different error.

Reply:
> I don't think Calcite can offer exact compatibility to Spark in this case.
> The best solution for Calcite is to give a runtime error.
> But I would still like to know why the date with a slash didn't report a different error.

@mihaibudiu Calcite will perform parser validation on the input parameters.
```
    final Calendar cal =
        DateTimeUtils.parseDateFormat(s, Format.get().date,
            DateTimeUtils.UTC_ZONE);
```
However, there is no restriction on the output result set. I've been researching for a long time and I'm not sure where to limit the result set

Reply:
I don't understand what you are saying.
If you add two very large numbers in Calcite you will get a runtime error.
How is this different?

Reply:
I mean, I don't know why I get a runtime error.
Sorry, I didn't express myself clearly.